apps/src/client.rs

```diff
@@ -485,15 +520,79 @@ pub fn connect(
             scid_sent = true;
         }
 
+        // Performs server side migration - client kicks it off.
         if args.perform_migration &&
-            !new_path_probed &&
+            !server_preferred_address_probed &&
             scid_sent &&
             conn.available_dcids() > 0
         {
+            // Probe the transport parameter's server preferred address.
+            if let Ok(Some(preferred_address_params)) =
+                conn.server_preferred_address_params()
+            {
+                let preferred_address_params = preferred_address_params.clone();
+
+                // Manually track the new Destination Connection ID and reset
+                // token of the preferred address.
+                conn.new_destination_cid(
+                    preferred_address_params.connection_id,
+                    1,
+                    preferred_address_params.stateless_reset_token,
+                    0,
+                    &mut SmallVec::new(),
+                )
+                .unwrap();
+
+                // Probe the path. If it successfully validates then the event
+                // handler will see the `PathEvent::Validated`
+                // event occur and begin migration. The server will see
+                // `PathEvent::PeerMigrated` upon successful
+                // migration.
+                if let Some(addr_v4) = preferred_address_params.addr_v4 {
```

Comment:
The selection of which preffered address to use is subjective - this is just an example. There's also bit of duplication across the branches that can be reduced. Consider extracting into a function like "select_and_probe_preferred_address"

Reply:
I coalesced the branches into one to dry the code.

Reply:
Lucas: "Put this into a function to make it more obvious to others"